tests/base64_tests.cpp

```diff
@@ -100,6 +101,94 @@ TEST(issue_520) {
   ASSERT_EQUAL(r.count, 0);
 }
 
+TEST(base64_decode_complete_input) {
+  std::vector<unsigned char> input_data = {0x54, 0x57, 0x46, 0x75}; // "TWFu" Base64 for "Man"
+  std::vector<char> expected_output = {'M', 'a', 'n'};
+  std::vector<char> output_buffer(expected_output.size());
+
+  // Test with all last_chunk_handling_options
+  for (auto option : {simdutf::last_chunk_handling_options::strict,
+                      simdutf::last_chunk_handling_options::loose,
+                      simdutf::last_chunk_handling_options::stop_before_partial}) {
+    auto result = implementation.base64_to_binary(
+        reinterpret_cast<const char*>(input_data.data()),
+        input_data.size(),
+        output_buffer.data(),
+        simdutf::base64_default,
+        option
+    );
+
+    ASSERT_EQUAL(result.error, simdutf::error_code::SUCCESS);
+    ASSERT_EQUAL(result.count, expected_output.size());
+    ASSERT_TRUE((std::equal(output_buffer.begin(),
+                           output_buffer.begin() + result.count,
+                           expected_output.begin())));
+  }
+}
+
+TEST(base64_decode_strict_mode) {
+  std::vector<std::pair<std::string, std::string>> test_cases = {
+    {"TQ", "M"},             // Length 2 (not multiple of 4)
+    {"TWE", "Ma"},           // Length 3 (not multiple of 4)
+    {"TWFu", "Man"},         // Length 4 (multiple of 4)
+    {"TWF1", "Mau"},         // Length 4 (multiple of 4)
+    {"TWFubWFu", "Manman"},  // Length 8 (multiple of 4)
+  };
+
+  for (const auto& [input_data, expected_output] : test_cases) {
```

Comment:
The simdutf library is still C++11.

So maybe something like this...

```C++
for (const auto& t : test_cases) {
  auto &input_data = t. input_data;
  auto & expected_output = t. expected_output;
...
}
```

or something of the sort! 

Sorry about the trouble.

Reply:
@FranciscoThiesen I am preparing a PR.

Reply:
Thanks for let me know, I can definitely adapt it in the next few days to C++11!

Reply:
Don't do anything. I will come up with a proposal.